### DISCUSSION TITLE
Preview with Neovim's syntax highlighting instead of bat

### DISCUSSION THREAD

**Discussion by psi4j** [gokcehan/lf#1729] [Q&A] [answered]
I've been trying to figure out how to work with Neovim's headless mode and hack a way to get it to provide syntax highlighting instead of using bat. I would like to have consistent syntax highlighting across my entire workspace and since I use Neovim, I thought I would try to find a way but I haven't been able to get it to work. 

The goal would be to let the LSP and Treesitter provide syntax highlighting for the preview. I was wondering if anyone here has done this or knows of a way to get this to work.

I'd appreciate hearing any ideas you guys may have because this would be an amazing way to sync everything seamlessly. Thank you.

**Comment by CatsDeservePets** [comment, marked as answer]
Hey @psi4j!

This is something I am interested as well. My way around this is using colour themes which also support `bat` themes (i.e. `.tmTheme` files; often under an `extras` directory). However, due to the way these themes work, some elements are still coloured differently. This is the closest I got with this.

**Comment by psi4j** [comment]
> Hey @psi4j!
> 
> 
> 
> This is something I am interested as well. My way around this is using colour themes which also support `bat` themes (i.e. `.tmTheme` files; often under an `extras` directory). However, due to the way these themes work, some elements are still coloured differently. This is the closest I got with this.

Hey there. I've actually switched over to using Yazi and use a custom bat theme myself. I appreciate you sharing this though. Somebody else might find this useful.

**Reply by CatsDeservePets** [reply]
Oh interesting (props to you for setting up a custom `bat` theme). I personally found setting up previews in `yazi` a little cumbersome and not as straight forward compared the `lf`. 

May I ask what made you switch over to `yazi`? The speed? Out of the box experience? Preferring `Rust` over `Go`?

Just trying to see what `lf` could do better.

**Reply by psi4j** [reply]
> May I ask what made you switch over to `yazi`?

I found that all the integrations I needed already worked out of the box with `yazi`. There's also the fact that it's written in Rust, which doesn't really matter a whole lot for it class of application, but because I'm a Rust developer, I am more open to contributing to `yazi` later if I find a need.

> Just trying to see what `lf` could do better.

I don't think there's anything wrong with `lf`. I've used it for years and found it is better than most its competitors. The only thing I think `yazi` has going for it is that it has many integrations out-of-the-box, meaning, the maintainers will maintain that code for the user. With regard to speed/usability/ergonomics, it's virtually the same.

Anyways, best of luck, my friend. I appreciate you responding to this post just in case.
